frame/offences/src/tests.rs

```diff
@@ -245,3 +281,157 @@ fn should_properly_count_offences() {
 		);
 	});
 }
+
+#[test]
+fn should_properly_store_offences() {
+	new_test_ext().execute_with(|| {
+		// given
+		let time_slot = 42;
+		let session_index = 10;
+		assert_eq!(offence_reports(KIND, time_slot), vec![]);
+
+		let offence1 =
+			Offence { validator_set_count: 5, session_index, time_slot, offenders: vec![5] };
+		let offence2 = Offence {
+			validator_set_count: 5,
+			session_index,
+			time_slot: time_slot + 1,
+			offenders: vec![4],
+		};
+		let offence3 = Offence {
+			validator_set_count: 5,
+			session_index: session_index + 1,
+			time_slot: time_slot + 1,
+			offenders: vec![6, 7],
+		};
+		let offence4 = Offence {
+			validator_set_count: 5,
+			session_index: session_index - 1,
+			time_slot: time_slot - 1,
+			offenders: vec![3],
+		};
+		Offences::report_offence(vec![], offence1).unwrap();
+		with_on_offence_fractions(|f| {
+			assert_eq!(f.clone(), vec![Perbill::from_percent(25)]);
+			f.clear();
+		});
+
+		// when
+		// report for the second time
+		Offences::report_offence(vec![], offence2).unwrap();
+		Offences::report_offence(vec![], offence3).unwrap();
+		Offences::report_offence(vec![], offence4).unwrap();
+
+		// then
+		let prev_session_reports = Vec::<(Kind, OpaqueTimeSlot, sp_core::H256)>::decode(
```

Comment:
You can def. use a helper function for this :D

Reply:
Also a type alias like `reportIdOf<T>` for `(Kind, OpaqueTimeSlot, sp_core::H256)`?